worker.js

```diff
@@ -0,0 +1,40 @@
+const proxyquire = require('proxyquire');
+const resolve = require('resolve');
+const serializeError = require('serialize-error');
+
+process.on('message', message => {
+	const {payload: {text, opts}} = message;
+	const {cwd, filename, config} = opts;
+
+	const xo = requireXO(cwd, config);
+	const report = xo.lintText(text, Object.assign({}, config || {}, {cwd, filename}));
+
+	process.send({
+		id: message.id,
+		payload: report
+	});
+});
+
+process.on('uncaughtException', err => {
+	process.send({error: serializeError(err)});
+});
+
+function tryResolveLocal(id, basedir) {
+	try {
+		return resolve.sync(id, {basedir});
```

Comment:
Use https://github.com/sindresorhus/resolve-from instead. It uses the Node.js resolve logic instead of a (buggy) reimplementation. Note that it returns `null` instead of throwing.